src/main/java/frc/robot/subsystems/Lift.java

```diff
@@ -137,15 +143,39 @@ public CommandBase moveTicks(double ticks, double speed) {
                 CommandScheduler.getInstance().requiring(this).cancel();
             }
         }, () -> {
-            return Math.abs(liftEncoder.getDistance()) >= ticks;
+            return Math.abs(liftEncoder.getDistance()) >= Math.abs(0.10);
         }, this);
-        cmd.setName("Move Ticks");
+        cmd.setName("Move Disengage");
         return cmd.withTimeout(0.2);
     }
 
+    public CommandBase moveTicks(double ticks, double speed) {
+        CommandBase cmd = new FunctionalCommand(() -> {
+            liftEncoder.reset();
+        }, () -> {
+            double realSpeed = speed;
+            if (ticks < 0 && speed > 0) {
+                realSpeed *= -1;
+            }
+
+            if (ticks < 0 && !upperLimitSwitch.getAsBoolean()) {
+                elevatorMotor.set(realSpeed);
+            } else if (ticks > 0 && !lowerLimitSwitch.getAsBoolean()) {
+                elevatorMotor.set(realSpeed);
+            }
+            SmartDashboard.putNumber("Lift Encoder", liftEncoder.getDistance());
+        }, (interrupted) -> {
+            elevatorMotor.stopMotor();
+        }, () -> {
+            return Math.abs(liftEncoder.getDistance()) >= Math.abs(ticks)
+                    || (ticks < 0 && upperLimitSwitch.getAsBoolean() || (ticks > 0 && lowerLimitSwitch.getAsBoolean()));
```

Comment:
Since `ticks` doesn't change within this function call, you could do those checks before creating the functional command, and use just the appropriate limit switch here.

```java
BooleanSupplier limit = () -> true;
if (ticks < 0) {
  limit = upperLimitSwitch::getAsBoolean;
} else if (ticks > 0) {
  limit = lowerLimitSwitch::getAsBoolean;
}
// ...
return Math.abs() /* ... */ || limit.getAsBoolean();
```

Reply:
resolvied